File: 프로그래머스/3/12979. 기지국 설치/기지국 설치.js
function solution(n, stations, w) {
    let pre = 0, ret =0;
    const len = w*2 +1;
    for(let i=0;i<stations.length;i++){
        let a = stations[i] - w - 1 - pre;
        let b = Math.ceil(a/len);
        ret += b;
        pre = stations[i] +w;
    }
    ret += Math.ceil( (n-pre) / len);
    return ret;
}

File: 프로그래머스/3/12979. 기지국 설치/기지국 설치.cpp
#include <iostream>
#include <vector>
#include <cmath>

using namespace std;
vector<int> ST; 
int N,W;
int go(int idx){
    if(idx >= N+1) return 0;
    
    bool flag = false;
    
    for(int st : ST){
        if (idx>= st-W && idx <= st + W ){
            idx = st;
            flag = true;
            break;
        }
    }
    
    if(flag) return go(idx+W+1);
    
    else return go(idx+2*W+1) + 1; 
    
}

int solution(int n, vector<int> stations, int w)
{
    N = n;
    ST =  stations;
    W = w;
    
    return go(1);
}